packages/nimble-components/src/anchor/tests/anchor.spec.ts

```diff
@@ -0,0 +1,292 @@
+import { DOM, html } from '@microsoft/fast-element';
+import {
+    DesignSystem,
+    Anchor as FoundationAnchor
+} from '@microsoft/fast-foundation';
+import { Anchor } from '..';
+import { fixture, Fixture } from '../../utilities/tests/fixture';
+
+async function setup(): Promise<Fixture<Anchor>> {
+    return fixture<Anchor>(html`<nimble-anchor></nimble-anchor>`);
+}
+
+describe('Anchor', () => {
+    let element: Anchor;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('should have its tag returned by tagFor(FoundationAnchor)', () => {
+        expect(DesignSystem.tagFor(FoundationAnchor)).toBe('nimble-anchor');
+    });
+
+    it('can construct an element instance', () => {
+        expect(document.createElement('nimble-anchor')).toBeInstanceOf(Anchor);
+    });
+
+    it('should set the "control" class on the internal control', async () => {
+        await connect();
+        expect(element.control.classList.contains('control')).toBe(true);
+    });
+
+    it('should set the `part` attribute to "control" on the internal control', async () => {
+        await connect();
+        expect(element.control.part.contains('control')).toBe(true);
+    });
+
+    it('should reflect `download` value to the internal control', async () => {
+        await connect();
+
+        element.setAttribute('download', 'filename');
+        await DOM.nextUpdate();
+
+        expect(element.control.getAttribute('download')).toBe('filename');
+    });
+
+    it('should reflect `hreflang` value to the internal control', async () => {
+        await connect();
+
+        element.setAttribute('hreflang', 'en');
+        await DOM.nextUpdate();
+
+        expect(element.control.getAttribute('hreflang')).toBe('en');
+    });
+
+    it('should reflect `ping` value to the internal control', async () => {
```

Comment:
Should we adopt a theory pattern for these tests?

Reply:
Good call. Done.